mlflow/tensorflow/_autolog.py

```diff
@@ -52,11 +59,162 @@ def on_epoch_end(self, epoch, logs=None):
         if epoch % self.log_every_n_steps == 0:
             self.metrics_logger.record_metrics(logs, epoch)
 
-    def on_train_end(self, logs=None):  # pylint: disable=unused-argument
-        if self.log_models:
-            registered_model_name = get_autologging_config(
-                mlflow.tensorflow.FLAVOR_NAME, "registered_model_name", None
-            )
-            mlflow.keras.log_model(
-                self.model, artifact_path="model", registered_model_name=registered_model_name
-            )
+
+def _extract_input_example_from_tensor_or_ndarray(
+    input_features: Union[tensorflow.Tensor, np.ndarray]
+) -> np.ndarray:
+    """
+    Extracts first `INPUT_EXAMPLE_SAMPLE_ROWS` from the next_input, which can either be of
+    numpy array or tensor type.
+
+    :param input_features: an input of type `np.ndarray` or `tensorflow.Tensor`
+    :return: a slice (of limit `INPUT_EXAMPLE_SAMPLE_ROWS`)  of the input of type `np.ndarray`
+
+    Examples
+    --------
+    when next_input is nd.array:
+    >>> input_data = np.array([1, 2, 3, 4, 5, 6, 7, 8])
+    >>> _extract_input_example_from_tensor_or_ndarray(input_data)
+    array([1, 2, 3, 4, 5])
+
+
+    when next_input is tensorflow.Tensor:
+    >>> input_data = tensorflow.convert_to_tensor([1, 2, 3, 4, 5, 6])
+    >>> _extract_input_example_from_tensor_or_ndarray(input_data)
+    array([1, 2, 3, 4, 5])
+    """
+
+    input_feature_slice = None
+    if isinstance(input_features, tensorflow.Tensor):
+        input_feature_slice = input_features.numpy()[0:INPUT_EXAMPLE_SAMPLE_ROWS]
+    elif isinstance(input_features, np.ndarray):
+        input_feature_slice = input_features[0:INPUT_EXAMPLE_SAMPLE_ROWS]
+    else:
+        warnings.warn(
+            f"Tensorflow estimator doesn't support '{type(input_features)}' type for features"
+        )
+    return input_feature_slice
+
+
+def _extract_sample_numpy_dict(
+    input_numpy_features_dict: Dict[str, np.ndarray]
+) -> Union[Dict[str, np.ndarray], np.ndarray]:
+    """
+    Extracts `INPUT_EXAMPLE_SAMPLE_ROWS` sample from next_input
+    as numpy array of dict(str -> ndarray) type.
+
+    :param input_numpy_features_dict: A tensor or numpy array
+    :return:a slice (limit `INPUT_EXAMPLE_SAMPLE_ROWS`)  of the input of same type as next_input
```

Comment:
Can we update this to also say that we return `None` if the type of `input_numpy_features_dict` is unsupported?